api/localizer/writer.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// See proposals/localize-command
+package localizer
+
+import (
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"sigs.k8s.io/kustomize/api/loader"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+const urlSeparator = "/"
+const localizeDir = "localized-files"
+const alpha = "alpha"
+
+var errNewDirRef = errors.New("target references newDir")
+var errAbsPath = errors.New("absolute paths not yet supported")
+
+func makeAndClean(path string, fSys filesys.FileSystem) (filesys.ConfirmedDir, error) {
+	var cleanPath filesys.ConfirmedDir
+	var err error
+
+	if err = fSys.MkdirAll(path); err != nil {
+		return "", errors.Wrapf(err, filesys.ErrMkDirAll.Error())
+	}
+	if cleanPath, _, err = fSys.CleanedAbs(path); err != nil {
+		return "", errors.Wrapf(err, filesys.ErrCleanAbs.Error())
+	}
+
+	return cleanPath, nil
+}
+
+// NewWriter is the factory method for Writer.
+func NewWriter(target filesys.ConfirmedDir, scope filesys.ConfirmedDir, newDir filesys.ConfirmedDir, fSys filesys.FileSystem) (*Writer, error) {
+	inNewDir, _ := filepath.Rel(scope.String(), target.String())
+	dst, dstErr := makeAndClean(newDir.Join(inNewDir), fSys)
+	if dstErr != nil {
+		return nil, errors.Wrapf(dstErr, "unable to write to newDir")
```

Comment:
```suggestion
		return nil, errors.Wrapf(dstErr, fmt.Sprintf("unable to write to new directory %q", newDir)
```

This should print something like `unable to write to new directory 'newDir'`

Reply:
I'm leaning towards not printing `newDir`'s path because it should already be included in `dstErr`

Reply:
I've re-written almost all of the error messages. Lmk if this is still a problem.